File: JavaToKotlin/app/src/main/java/io/dustin/apps/board/domain/community/comment/service/ReadCommentService.java
package io.dustin.apps.board.domain.community.comment.service;

import io.dustin.apps.board.domain.community.comment.model.Comment;
import io.dustin.apps.board.domain.community.comment.model.dto.CommentDto;
import io.dustin.apps.board.domain.community.comment.repository.CommentRepository;
import io.dustin.apps.common.exception.DataNotFoundException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

/**
 * cqrs -> srp
 */
@Service
@RequiredArgsConstructor
public class ReadCommentService {

    private final CommentRepository commentRepository;

    @Transactional(readOnly = true)
    public List<CommentDto> getCommentsByPosting(long userId, long postingId, int size, Long nextId) {
        return commentRepository.commentListByPosting(userId, postingId, size, nextId);
    }

    @Transactional(readOnly = true)
    public List<CommentDto> replyListByComment(long userId, long commentId, int size, Long nextId) {
        return commentRepository.replyListByComment(userId, commentId, size, nextId);
    }


    @Transactional(readOnly = true)
    public Comment findById(long commentId) {
        Optional<Comment> optional = this.commentRepository.findById(commentId);
        if(optional.isPresent()) {
            return optional.get();
        } else {
            return null;
        }
    }
    @Transactional(readOnly = true)
    public Comment findByIdOrThrow(long commentId) {
        Optional<Comment> optional = this.commentRepository.findById(commentId);
        if(optional.isPresent()) {
            return optional.get();
        } else {
            throw new DataNotFoundException("""
                    id [#1]로 조회된 게시물이 없습니다.""".replace("#1", String.valueOf(commentId)));
        }
    }

}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/community/comment/service/ReadCommentService.kt
package io.dustin.apps.board.domain.community.comment.service

import io.dustin.apps.board.domain.community.comment.model.Comment
import io.dustin.apps.board.domain.community.comment.repository.CommentRepository
import io.dustin.apps.board.domain.qna.answer.model.Answer
import io.dustin.apps.common.exception.DataNotFoundException
import io.dustin.apps.common.repository.findByIdOrThrow
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service

@Service
class ReadCommentService (
    private val commentRepository: CommentRepository
) {

    /**
     * 게시물에 대한 댓글 리스트 불러오기
     */
    fun getCommentsByPosting(userId: Long, postingId: Long, recordsCount: Long, nextId: Long?) =
        commentRepository.commentListByPosting(userId, postingId, recordsCount, nextId)

    /**
     * 하위댓글 리스트 불러오기
     */
    fun replyListByComment(userId: Long, commentId: Long, recordsCount: Long, nextId: Long?) =
        commentRepository.replyListByComment(userId, commentId, recordsCount, nextId)

    fun findByIdOrNull(commentId: Long): Comment? {
        return commentRepository.findByIdOrNull(commentId)
    }

    fun findByIdOrThrow(commentId: Long): Comment {
        return commentRepository.findByIdOrThrow(commentId)
    }
}
